octoprint_Spoolman/static/js/Spoolman_modal_selectSpool.js

```diff
@@ -150,6 +150,7 @@ $(() => {
                 id: true,
                 spoolName: true,
                 material: true,
+                lot: true,
```

Comment:
To move forward with this functionality, I think it has to become configurable via the plugin's settings panel. For people like myself, who don't print that much, using lot number is completely unnecessary, therefore this column will just waste screen space to display no value.

Some of the infrastructure to show / hide columns is already there, so hopefully this will boil down to just adding a checkbox in the settings and that should be all. Ideally, all of the currently present columns should be toggleable, but as that's not the main focus of this PR, I'm fine with introducing it for just this new column.

Reply:
I will take a look at how the settings integration works, truthfully this is the first Octoprint plugin I have worked on. I will take a look at Octoprint's documentation, but any pointers for your implementation would be appreciated!

I would obviously need to add a section to `Spoolman_settings.jinja2`, but does it need to be in a new tab, for example `Display Settings`?

Reply:
I was able to get the setting working with it defaulting to false, however changing the setting requires a page reload before it takes effect. I imagine there is a way to force that update, possibly in `Spoolman_settings.js`? Here is the updated code for the table visibility for reference.
![image](https://github.com/user-attachments/assets/9364ab71-6732-4482-8fa8-0ba3796de6e0)

Reply:
> I would obviously need to add a section to `Spoolman_settings.jinja2`, but does it need to be in a new tab, for example `Display Settings`?

Yes, that's correct, that's where all settings reside. Creating a new tab for the display settings is a good idea, so feel free to explore that, but I won't block the PR if you won't be able to implement it that way.

Reply:
> I was able to get the setting working with it defaulting to false, however changing the setting requires a page reload before it takes effect. I imagine there is a way to force that update, possibly in `Spoolman_settings.js`? Here is the updated code for the table visibility for reference.

Octoprint uses the Knockout framework which implements [observables](https://knockoutjs.com/documentation/observables.html) to handle automatic updates like the one you would like to see. I'm not super versed in that framework myself, but I think you would have to turn `tableAttributeVisibility` variable into an observable (maybe `ko.computed`) to auto update on settings change. To be fair, I don't know how Knockout "knows" that a dependency of such computed observable has changed, but you could give it a try, there should be at least one usage of `ko.computed` somewhere in the codebase.